library/std/src/pipe.rs

```diff
@@ -41,6 +127,37 @@ impl PipeReader {
 
 impl PipeWriter {
     /// Create a new [`PipeWriter`] instance that shares the same underlying file description.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(anonymous_pipe)]
+    /// # #[cfg(miri)] fn main() {}
+    /// # #[cfg(not(miri))]
+    /// # use std::process::Command;
+    /// # use std::io::Read;
+    /// # fn main() -> std::io::Result<()> {
+    /// let (mut reader, writer) = std::pipe::pipe()?;
+    ///
+    /// let mut peer = Command::new("python")
+    /// .args([
+    ///     "-c",
+    ///     "from sys import stdout, stderr\n\
+    ///      stdout.write('foo')\n\
+    ///      stderr.write('bar')"
+    /// ])
```

Comment:
For this I feel like bash is simpler and most people can under it pretty well, and it could run on playground:

```suggestion
    /// let mut peer = Command::new("bash")
    /// .args([
    ///     "-c",
    ///     "echo foo\n\
    ///      echo bar >&2"
    /// ])
```